src/librustc_borrowck/borrowck/mir/elaborate_drops.rs

```diff
@@ -586,6 +586,11 @@ impl<'b, 'tcx> ElaborateDropsCtxt<'b, 'tcx> {
                             // drop elaboration should handle that by itself
                             continue
                         }
+                        TerminatorKind::Resume => {
+                            // We can replace resumes with gotos
+                            // jumping to a canonical resume.
```

Comment:
My hypothesis is that something I did here is causing changes to compiler control flow overall so that the `patch.patch_map` now holds entries for one or more resume terminators (replacing the resumes with gotos). Before this change, we never exercised that aspect of the patching code, and so this iteration here would fall into the `_ => { ... }` block where it asserts `!self.patch.is_patched(bb)`. But in this case, the code *is* patched (with the aforementioned replacement).

That's my current hypothesis / reconstructed memory.

Working on confirming this, and figuring out a clearer way to explain it. (Not sure if its worth trying to avoid it, since the patch code itself does *seem* to support this scenario, even if its been untested...)